src/normalize.js

```diff
@@ -92,21 +94,65 @@ function extractAllLinks(json, uri, opts) {
   return ret;
 }
 
+function extractToVirtualKey(uri, rel, content, contentKey) {
+  const ret = {};
+  const virtualKey = `${uri}#${rel}`;
+  ret[virtualKey] = {
+    [contentKey]: content,
+  };
+
+  ret[uri] = {};
+  ret[uri][rel] = {
+    href: virtualKey,
+    virtual: true,
+  };
+  return ret;
+}
+
 function mergeEmbeddedStandaloneCollections(embedded, links, opts) {
   const ret = {};
   merge(ret, links);
   merge(ret, embedded);
 
   keys(embedded).forEach((uri) => {
+    // check all embedded properties for embedded collections
     keys(embedded[uri]).forEach((rel) => {
-      if (Array.isArray(embedded[uri][rel]) && uri in links && rel in links[uri]) {
-        ret[uri][rel] = links[uri][rel];
-        ret[links[uri][rel].href] = {
-          [opts.embeddedStandaloneListKey]: embedded[uri][rel],
-          [opts.metaKey]: { self: links[uri][rel].href },
-        };
+      if (Array.isArray(embedded[uri][rel])) {
+        // standalone link provided (store embedded list as standalone link)
+        if (uri in links && rel in links[uri] && isSingleLink(links[uri][rel])) {
+          ret[uri][rel] = links[uri][rel];
+          ret[links[uri][rel].href] = {
+            [opts.embeddedStandaloneListKey]: embedded[uri][rel],
+            [opts.metaKey]: { self: links[uri][rel].href },
+          };
+        } else if (opts.embeddedStandaloneListVirtualKeys
+          && rel !== opts.embeddedStandaloneListKey) {
+          // no standalone link provided --> generate virtual key
+          delete ret[uri][rel];
+          merge(
+            ret,
+            extractToVirtualKey(uri, rel, embedded[uri][rel], opts.embeddedStandaloneListKey),
+          );
+        }
       }
     });
+
+    // also check remaining link properties to search for a possible collection
+    // which is not embedded
+    if (opts.embeddedStandaloneListVirtualKeys) {
+      difference(
+        keys(links[uri]),
+        [...keys(embedded[uri]), opts.embeddedStandaloneListKey],
```

Comment:
Why do we need an exception for a link with this name here and on line 129? I thought we didn't even support a link rel called "items", or if the API really used this rel, one should use a different embeddedStandaloneListKey..?

Reply:
Well, the API uses it for collections. API platform returns something like this:
```javascript
{
  "_links": {
    "self": {
      "href": "/activities"
    },
    "items": [
      {
        "href": "/activities/b3acd3cf5282"
      }
    ]
  },
  "totalItems": 5,
  "_embedded": {
    "items": [
      {
        "_links": {
          "self": {
            "href": "/activities/b3acd3cf5282"
          },
        },
        "title": "Neue Aktivität",
        "location": "test",
        "id": "b3acd3cf5282"
      }
    ]
  }
}
```

Without excluding items, this would extract #items from the collection and normalize to something like this:
```javascript
{
    "/activities": {
        "totalItems": 5,
        "items": {
            "href": "/activities#items",
            "virtual": true
        }
    },
    "/activities#items": {
        "items": [
            {
                "href": "/activities/b3acd3cf5282"
            }
        ]
    },
    "/activities/b3acd3cf5282": {
        "title": "Neue Aktivität",
        "location": "test",
        "id": "b3acd3cf5282"
    }
}
```

So normally (at least for us) `embeddedStandaloneListKey` matches the same key that is used to transmit collection items. However, the hal standard doesn't specify this AFAIK. If we want to make this more flexible, we could separate this 2 options and allow to configure an array of property names that is always excluded from generating virtual links.